cosmos/operators/local.py

```diff
@@ -234,58 +234,26 @@ def store_compiled_sql(self, tmp_project_dir: str, context: Context) -> None:
-        @provide_session
-        def _store_compiled_sql(session: Session = NEW_SESSION) -> None:
-
-            if not self.should_store_compiled_sql:
-                return
-
-            compiled_queries = {}
-            # dbt compiles sql files and stores them in the target directory
-            for folder_path, _, file_paths in os.walk(os.path.join(tmp_project_dir, "target")):
-                for file_path in file_paths:
-                    if not file_path.endswith(".sql"):
-                        continue
-
-                    compiled_sql_path = Path(os.path.join(folder_path, file_path))
-                    compiled_sql = compiled_sql_path.read_text(encoding="utf-8")
-
-                    relative_path = str(compiled_sql_path.relative_to(tmp_project_dir))
-                    compiled_queries[relative_path] = compiled_sql.strip()
-
-            for name, query in compiled_queries.items():
-                self.compiled_sql += f"-- {name}\n{query}\n\n"
+        compiled_queries = {}
+        # dbt compiles sql files and stores them in the target directory
+        for folder_path, _, file_paths in os.walk(os.path.join(tmp_project_dir, "target")):
+            for file_path in file_paths:
+                if not file_path.endswith(".sql"):
+                    continue
 
-            self.compiled_sql = self.compiled_sql.strip()
+                compiled_sql_path = Path(os.path.join(folder_path, file_path))
+                compiled_sql = compiled_sql_path.read_text(encoding="utf-8")
 
-            # need to refresh the rendered task field record in the db because Airflow only does this
-            # before executing the task, not after
-            from airflow.models.renderedtifields import RenderedTaskInstanceFields
+                relative_path = str(compiled_sql_path.relative_to(tmp_project_dir))
+                compiled_queries[relative_path] = compiled_sql.strip()
 
-            ti = context["ti"]
-
-            if isinstance(
-                ti, TaskInstance
-            ):  # verifies ti is a TaskInstance in order to access and use the "task" field
-                if TYPE_CHECKING:
-                    assert ti.task is not None
-                ti.task.template_fields = self.template_fields
-                rtif = RenderedTaskInstanceFields(ti, render_templates=False)
-
-                # delete the old records
-                session.query(RenderedTaskInstanceFields).filter(
-                    RenderedTaskInstanceFields.dag_id == self.dag_id,  # type: ignore[attr-defined]
-                    RenderedTaskInstanceFields.task_id == self.task_id,
-                    RenderedTaskInstanceFields.run_id == ti.run_id,
-                ).delete()
-                session.add(rtif)
-            else:
-                self.log.info("Warning: ti is of type TaskInstancePydantic. Cannot update template_fields.")
+        for name, query in compiled_queries.items():
+            self.compiled_sql += f"-- {name}\n{query}\n\n"
```

Comment:
This is so much cleaner, super happy to see it coming through 🎉

Reply:
yes, I am glad too :)